From b7f87e9e290bd465570259345c039c0df57141ee Mon Sep 17 00:00:00 2001
From: klzgrad <kizdiv@gmail.com>
Date: Thu, 7 Aug 2025 22:22:46 +0800
Subject: [PATCH] third_party/perfetto: Add libperfetto when
 build_with_chromium=false


Adapted to Chromium 156 (additional whitespace in the removed block).

--- a/src/third_party/perfetto/BUILD.gn
+++ b/src/third_party/perfetto/BUILD.gn
@@ -301,7 +301,7 @@
 # "libperfetto_client_experimental" targets defined within this BUILD.gn file.
 # Rationalize them with eseckler@. For now seems this one is only used from
 # chromium and the other one only from the Android tree.
-if (build_with_chromium) {
+if (true) {
   component("libperfetto") {
     public_configs = [ "gn:public_config" ]
     deps = [
@@ -343,34 +343,7 @@
     }
   }
 
-  # TODO(altimin): this is a temp workaround around very strange link failures
-  # on win-dbg-ng buildbot.
-  if (!is_win) {
-    libtrace_processor_target_type = "source_set"
-  } else {
-    libtrace_processor_target_type = "component"
-  }
-
-  # In Chromium, we want to ensure that we don't link dynamically against sqlite
-  # (as Chromium also uses a more restricted version of sqlite which is actually
-  # shipped to the users).
-  # source_set helps us to achieve that.
-  target(libtrace_processor_target_type, "libtrace_processor") {
-    public_configs = [ "gn:public_config" ]
-    deps = [ "src/trace_processor:lib" ]
-    configs -= [ "//build/config/compiler:chromium_code" ]  # nogncheck
-    configs += [ "//build/config/compiler:no_chromium_code" ]  # nogncheck
-    public_deps = [ "include/perfetto/trace_processor" ]
-  }
-
-  component("perfetto_test_support") {
-    testonly = true
-    public_configs = [ "gn:public_config" ]
-    configs -= [ "//build/config/compiler:chromium_code" ]  # nogncheck
-    configs += [ "//build/config/compiler:no_chromium_code" ]  # nogncheck
-    public_deps = [ "include/perfetto/test:test_support" ]
-    deps = [ "src/tracing/test:test_support" ]
-  }
+
 }
 
 # +----------------------------------------------------------------------------+
